prometheus_client/__init__.py

```diff
@@ -446,6 +447,82 @@ def write_to_textfile(path, registry):
     os.rename(tmppath, path)
 
 
+
+class ProcessCollector(object):
+    """Collector for Standard Exports such as cpu and memory."""
+    def __init__(self, namespace='', pid='self', proc='/proc', registry=REGISTRY):
+        self._namespace = namespace
+        self._pid = os.path.join(proc, str(pid))
+        self._proc = proc
+        self._pagesize = resource.getpagesize()
+        if namespace:
+            self._prefix = namespace + '_process_'
+        else:
+            self._prefix = 'process_'
+        self._ticks = 100.0
+        try:
+            self._ticks = os.sysconf('SC_CLK_TCK')
+        except (ValueError, TypeError):
+            pass
+
+        self._can_read_proc = os.access(os.path.join(self._proc, 'stat'), os.R_OK)
+        if self._can_read_proc:
+            self._btime = self._boot_time()
+        if registry:
+          registry.register(self)
+
+    def _boot_time(self):
+        with open(os.path.join(self._proc, 'stat')) as stat:
+            for line in stat:
+                if line.startswith('btime '):
+                    return float(line.split()[1])
+
+    def collect(self):
+        if not self._can_read_proc:
+            return []
+
+        result = []
+        try:
+            with open(os.path.join(self._pid, 'stat')) as stat:
+                parts = (stat.read().split(')')[-1].split())
+            vmem = Metric(self._prefix + 'virtual_memory_bytes', 'Virtual memory size in bytes', 'gauge')
+            vmem.add_sample(self._prefix + 'virtual_memory_bytes', {}, float(parts[20]))
+            rss = Metric(self._prefix + 'resident_memory_bytes', 'Resident memory size in bytes', 'gauge')
+            rss.add_sample(self._prefix + 'resident_memory_bytes', {}, float(parts[21]) * self._pagesize)
+            start_time = Metric(self._prefix + 'start_time_seconds',
+                                'Start time of the process since unix epoch in seconds.', 'gauge')
+            start_time_secs = float(parts[18]) / self._ticks
```

Comment:
This doesn't seem right it appears to be always returning 0, I think it should be 19